noxfile.py

```diff
@@ -92,10 +125,41 @@ def default(session, install_grpc=True):
         "pytest-xdist",
     )
 
-    if install_grpc:
-        session.install("-e", ".[grpc]", "-c", constraints_path)
+    constraints_dir = str(CURRENT_DIRECTORY / "testing")
+
+    if prerelease:
+        install_prerelease_dependencies(
+            session, f"{constraints_dir}/constraints-{PYTHON_VERSIONS[0]}.txt"
```

Comment:
I'm confused: in line 206, where we call `default(..., prerelease=True)`, the decorator constrains `python=PYTHON_VERSIONS[-1]`. But here we are referencing the constraints for  `PYTHON_VERSIONS[0]`. Could you clarify?

Reply:
As per the comment in below, the constraints file for the lowest supported python version contains a list of all of the dependencies of the library. The function `install_prerelease_dependencies` will extract the dependencies  from the file.

https://github.com/googleapis/python-api-core/blob/126b5c7154223da5f0537c90462a49e7e0d3ad97/testing/constraints-3.7.txt#L1-L7

https://github.com/googleapis/python-api-core/blob/7fbce0dfe60502f4b9e8502f2bef11dc9a2c551b/noxfile.py#L79-L88

Once we have a list of dependencies, we will install them independently with the `--pre` and `--no-deps` option. `--upgrade is also added to ensure that we get the latest pre-release.

https://github.com/googleapis/python-api-core/blob/7fbce0dfe60502f4b9e8502f2bef11dc9a2c551b/noxfile.py#L99-L100